libsql-remote/src/lib.rs

```diff
@@ -0,0 +1,290 @@
+        }
+        Ok(affected_row_count)
+    }
+
+    pub async fn query(&mut self, params: &Params) -> Result<Rows> {
+        let mut stmt = self.inner.clone();
+        bind_params(params.clone(), &mut stmt);
+
+        let StmtResult { rows, cols, .. } = self.client.execute_inner(stmt, 0).await?;
+
+        Ok(Rows {
+            rows,
+            cols: Arc::new(cols),
+        })
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Rows {
+    cols: Arc<Vec<Col>>,
+    rows: VecDeque<Vec<Value>>,
+}
+
+impl Serialize for Rows {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        struct ColsRef<'a>(&'a [Col]);
+        impl<'a> Serialize for ColsRef<'a> {
+            fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+            where
+                S: Serializer,
+            {
+                let mut s = serializer.serialize_seq(Some(self.0.len()))?;
+                for col in self.0.iter() {
+                    s.serialize_element(&col.name)?;
+                }
+                s.end()
+            }
+        }
+
+        struct RowsRef<'a>(&'a VecDeque<Vec<Value>>);
+        impl<'a> Serialize for RowsRef<'a> {
+            fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+            where
+                S: Serializer,
+            {
+                let mut s = serializer.serialize_seq(Some(self.0.len()))?;
+                for row in self.0.iter() {
+                    s.serialize_element(&RowRef(row))?;
+                }
+                s.end()
+            }
+        }
+        struct RowRef<'a>(&'a [Value]);
+        impl<'a> Serialize for RowRef<'a> {
+            fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+            where
+                S: Serializer,
+            {
+                let mut s = serializer.serialize_seq(Some(self.0.len()))?;
+                for cell in self.0.iter() {
+                    match cell {
+                        Value::Null => s.serialize_element(&Value::Null),
+                        Value::Integer { value } => s.serialize_element(value),
+                        Value::Float { value } => s.serialize_element(value),
+                        Value::Text { value } => s.serialize_element(value),
+                        Value::Blob { value } => {
+                            let base64 = STANDARD_NO_PAD.encode(value);
+                            s.serialize_element(&json!({
+                                "base64": base64
+                            }))
+                        }
+                    }?;
+                }
+                s.end()
+            }
+        }
+
+        let mut s = serializer.serialize_struct("Rows", 2)?;
+        s.serialize_field("columns", &ColsRef(self.cols.deref()))?;
+
+        s.end()
+    }
+}
+
+impl Iterator for Rows {
```

Comment:
We choose not to use Iterator for rows for a few reasons so we should follow what we do in libsql.